Sources/App/Core/AppEnvironment.swift

```diff
@@ -26,6 +26,8 @@ struct AppEnvironment {
     var rollbarLogLevel: () -> AppError.Level
     var shell: Shell
     var siteURL: () -> String
+    var twitterCredentials: () -> Twitter.Credentials?
+    var twitterPostTweet: (_ client: Client, _ tweet: String) -> EventLoopFuture<Void>
```

Comment:
We may also want to add an off switch to disable tweeting without having to deploy a new release. Or we just yank the credentials - but switching it on again is then more fiddly, because we need to look them up and enter them again, which is error prone.

Reply:
Yea, I think this is a great idea. Let's do that before we push it live so we can switch it off with an environment change, or some other switch.